web/src/pages/EditCustomLog/EditCustomLog.tsx

```diff
@@ -0,0 +1,141 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+import React from 'react';
+import { Alert, Link, Text, useSnackbar } from 'pouncejs';
+import Panel from 'Components/Panel';
+import CustomLogForm from 'Components/forms/CustomLogForm';
+import { CUSTOM_LOG_TYPES_DOC_URL } from 'Source/constants';
+import { ErrorCodeEnum } from 'Generated/schema';
+import { extractErrorMessage } from 'Helpers/utils';
+import {
+  EventEnum,
+  SrcEnum,
+  trackError,
+  TrackErrorEnum,
+  trackEvent,
+  PageViewEnum,
+} from 'Helpers/analytics';
+import { compose } from 'Helpers/compose';
+import Page404 from 'Pages/404';
+import withSEO from 'Hoc/withSEO';
+import useRouter from 'Hooks/useRouter';
+
+import useTrackPageView from 'Hooks/useTrackPageView';
+import { useUpdateCustomLog } from './graphql/updateCustomLog.generated';
+import { useGetCustomLogDetails } from '../CustomLogDetails/graphql/getCustomLogDetails.generated';
+
+const EditCustomLog: React.FC = () => {
+  useTrackPageView(PageViewEnum.CustomLogEditing);
+
+  const { match: { params: { logType } } } = useRouter<{ logType: string }>(); // prettier-ignore
+  const { pushSnackbar } = useSnackbar();
+  const { data, error: uncontrolledError } = useGetCustomLogDetails({
+    variables: { input: { logType } },
+  });
+
+  const [updateCustomLog] = useUpdateCustomLog({
+    onCompleted: ({ updateCustomLog: { error } }) => {
+      if (!error) {
+        trackEvent({ event: EventEnum.UpdatedCustomLog, src: SrcEnum.CustomLogs });
+        pushSnackbar({
+          variant: 'success',
+          title: 'Successfully updated custom log schema!',
+        });
+      } else {
+        trackError({ event: TrackErrorEnum.FailedToUpdateLogSource, src: SrcEnum.CustomLogs });
+        pushSnackbar({ variant: 'error', title: error.message });
+      }
+    },
+    onError: error => {
+      trackError({ event: TrackErrorEnum.FailedToAddCustomLog, src: SrcEnum.CustomLogs });
+      pushSnackbar({
+        variant: 'error',
+        title: extractErrorMessage(error),
+      });
+    },
+  });
+
+  const { record: customLog, error: controlledError } = data?.getCustomLog || {};
+
+  const initialValues = React.useMemo(
+    () => ({
+      revision: data?.getCustomLog?.record?.revision,
+      name: data?.getCustomLog?.record?.logType ?? 'Loading...',
+      referenceUrl: data?.getCustomLog?.record?.referenceURL ?? 'Loading...',
+      schema: data?.getCustomLog?.record?.logSpec ?? 'Loading...',
+      description: data?.getCustomLog?.record?.description ?? 'Loading...',
+    }),
+    [data]
+  );
+
+  if (data?.getCustomLog?.error) {
```

Comment:
```suggestion
  if (controlledError) {
```

I think that's what you originally intended to write, since - if we keep what you had - the condition is duplicated right below